django/src/rdwatch/views/server_status.py

```diff
@@ -1,70 +1,35 @@
 import datetime
 import socket
-from dataclasses import dataclass
 
-from rest_framework import permissions, serializers
-from rest_framework.response import Response
-from rest_framework.schemas.openapi import AutoSchema
-from rest_framework.views import APIView
-
-SERVER_INSTANCE_EPOCH = datetime.datetime.now()
-
-
-@dataclass
-class ServerStatus:
-    uptime: datetime.timedelta
-    hostname: str
-    ip: str
+from ninja import Field, Router, Schema
 
+from django.http import HttpRequest
 
-class UptimeSerializer(serializers.Serializer):
-    iso8601 = serializers.DurationField(source='*')
-    days = serializers.IntegerField()
-    seconds = serializers.IntegerField()
-    useconds = serializers.IntegerField(source='microseconds')
+router = Router()
 
+SERVER_INSTANCE_EPOCH = datetime.datetime.now()
 
-class ServerStatusSerializer(serializers.Serializer):
-    uptime = UptimeSerializer(required=True)
-    hostname = serializers.CharField(required=True)
-    ip = serializers.IPAddressField(required=True)
-
-    def create(self, validated_data):
-        instance = ServerStatus(
-            uptime=validated_data['uptime']['iso8601'],
-            hostname=validated_data['hostname'],
-            ip=validated_data['iso8601'],
-        )
-        # we'd save it here if this were a Django model
-        # instance.save()
-        return instance
-
-    def update(self, instance, validated_data):
-        instance.uptime = validated_data.get('uptime', instance.uptime)
-        instance.hostname = validated_data.get('hostname', instance.hostname)
-        instance.ip = validated_data.get('ip', instance.ip)
-        # we'd save it here if this were a Django model
-        # instance.save()
-        return instance
 
+class UptimeSchema(Schema):
+    iso8601: str
+    days: int
+    seconds: int
+    useconds: int = Field(..., alias='microseconds')
 
-class ServerStatusSchema(AutoSchema):
-    def get_operation_id(self, *args):
-        return 'getStatus'
+    @staticmethod
+    def resolve_iso8601(obj: datetime.timedelta):
+        return str(obj)
```

Comment:
Note, this returns a slightly different string than the DRF serializer for `datetime.timedelta`. Previously, the return value would look something like `6 05:04:40.740791`, but now it is `6 days, 05:04:40.740791`.

Reply:
In testing, I'm getting back an iso8601 duration format:  `"iso8601": "P0DT00H03M56.226221S"` which I think is fine.